### ISSUE TITLE
Release spanishoddata 0.1.0

### ISSUE THREAD

**Issue by e-kotov** [rOpenSpain/spanishoddata#115]
First release:

* [x] `usethis::use_cran_comments()`
* [x] Update (aspirational) install instructions in README
* [x] Proofread `Title:` and `Description:`
* [x] Check that all exported functions have `@return` and `@examples`
* [x] Check that `Authors@R:` includes a copyright holder (role 'cph')
* [x] Check [licensing of included files](https://r-pkgs.org/license.html#sec-code-you-bundle)
* [x] Review <https://github.com/DavisVaughan/extrachecks>

Prepare for release:

* [x] `git pull`
* [x] `urlchecker::url_check()`
* [x] `devtools::check(remote = TRUE, manual = TRUE)`
* [x] `devtools::check_win_devel()`
* [x] `git push`

Submit to CRAN:

* [x] `usethis::use_version('patch')`
* [x] `devtools::submit_cran()`
* [x] Approve email

Wait for CRAN...

* [x] Accepted :tada:
* [x] `usethis::use_github_release()`
* [ ] `usethis::use_dev_version(push = TRUE)`

**Comment by e-kotov** [comment]
CRAN review:

```
Please add \value to .Rd files regarding exported methods and explain
the functions results in the documentation. Please write about the
structure of the output (class) and also what the output means. (If a
function does not return a value, please document that too, e.g.
\value{No return value, called for side effects} or similar)
For more details:
<https://contributor.r-project.org/cran-cookbook/docs_issues.html#missing-value-tags-in-.rd-files>
Missing Rd-tags:
      spod_disconnect.Rd: \value

You have examples for unexported functions. Please either omit these
examples or export these functions.
Examples for unexported function
   spod_available_data_v1() in:
      spod_available_data_v1.Rd
   spod_available_data_v2() in:
      spod_available_data_v2.Rd
   spod_get_latest_v1_file_list() in:
      spod_get_latest_v1_file_list.Rd
   spod_get_latest_v2_file_list() in:
      spod_get_latest_v2_file_list.Rd
   spod_get_zones_v1() in:
      spod_get_zones_v1.Rd
   spod_get_zones_v2() in:
      spod_get_zones_v2.Rd

Some code lines in examples are wrapped in if(FALSE). Please never do
that. Ideally find toy examples that can be regularly executed and
checked. Lengthy examples (> 5 sec), can be wrapped in \donttest{}.

\dontrun{} should only be used if the example really cannot be executed
(e.g. because of missing additional software, missing API keys, ...) by
the user. That's why wrapping examples in \dontrun{} adds the comment
("# Not run:") as a warning for the user. Does not seem necessary.
Please replace \dontrun with \donttest.
Please put functions which download data in \donttest{}.
For more details:
<https://contributor.r-project.org/cran-cookbook/general_issues.html#structuring-of-examples>


Please fix and resubmit.
```


Will fix and resubmit soon.

**Comment by Robinlovelace** [comment]
Pretty good for a first submission. There are pros of CRAN's stringency.

**Comment by e-kotov** [comment]
> There are pros of CRAN's stringency.

There definitely are. I missed quite a few things in documentation, because I focused too much on the vignettes, so I had a feeling that the functions are well documented.

I wish these simple things were caught early with R CMD check or some other basic tools in 'devtools' or 'usethis'.

**Comment by Robinlovelace** [comment]
Yes, surely more of it could be automated, saving CRAN employees and developers time.

**Comment by e-kotov** [comment]
Resubmitted.

**Comment by e-kotov** [comment]
It's on the ~AppStore~CRAN https://cran.r-project.org/package=spanishoddata.

@Robinlovelace Do we want to have the dev branch moving forward? This would allows us to have the equivalent of latest CRAN release published on r-universe and keep the dev in a separate space.

**Comment by Robinlovelace** [comment]
> @Robinlovelace Do we want to have the dev branch moving forward? This would allows us to have the equivalent of latest CRAN release published on r-universe and keep the dev in a separate space.

My advice would be not to bother with a roling "dev" branch. Why? Because the maintenance burden is not worthwhile for such a small project with few contributors. I have contributed to a project that had this approach, also called "Git workflow", and saw little benefit. It seems now that the `develop` branch in that project is no longer maintained: https://github.com/luukvdmeer/sfnetworks/branches

I'm a fan of keeping things simple and GitHub workflow, with the main version being dev and releases used to signify stable versions, does that nicely. See here for pros and cons on git-flow vs github-flow: https://stackoverflow.com/questions/18188492/what-are-the-pros-and-cons-of-git-flow-vs-github-flow

> Start with a model as simple as possible (like GitHub flow tends to be), and move towards a more complex model if you need to.

is good advice.

**Comment by e-kotov** [comment]
> If people opt in to install from GitHub or elsewhere they know they're getting the dev version is the convention for most R packages, right?

Kind of, yes. For now. You're right, r-universe is not as popular as CRAN. I will however note in the readme that the install instruction from r-universe will get them the current dev version.

**Comment by Robinlovelace** [comment]
```
usethis::use_dev_version(push = TRUE)
```

Does not create a new branch, it just increments the version number.

**Comment by e-kotov** [comment]
> Does not create a new branch, it just increments the version number.

should we do it though?

**Comment by e-kotov** [comment]
ahh, I see now #122
